src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas3/DomainEventService.kt

```diff
@@ -209,8 +214,8 @@ class DomainEventService(
         createdAt = OffsetDateTime.now(),
         data = objectMapper.writeValueAsString(domainEvent.data),
         service = "CAS3",
-        triggerSource = null,
-        triggeredByUserId = null,
+        triggeredByUserId = user?.id,
+        triggerSource = if (user != null) TriggerSourceType.USER else TriggerSourceType.SYSTEM,
```

Comment:
in CAS1 it's possible for a domain event that was triggered via an API request for a specific user to still be classed as a SYSTEM event.

For example, when an application is submitted we sometimes automatically allocate the application's assessment to a user. This 'assessment allocated' domain event will have a TriggerSourceType of SYSTEM, despite triggeredByUserId being set to a value (i.e. the user didn't decide to allocate the assessment to someone, so it would be wrong to show their name next to the domain event)

For consistency, i'd recommend that TriggerSourceType is only ever set explicitly by the calling code (i.e. set to null unless calling code says otherwise). UI can then make the assumption that if source type is null the user should be shown (which it would have to do anyway to deal with existing domain events where this isn't set).